packages/inference/src/providers/novita.ts

```diff
@@ -50,38 +51,88 @@ export class NovitaConversationalTask extends BaseConversationalTask {
 		return "/v3/openai/chat/completions";
 	}
 }
+
 export class NovitaTextToVideoTask extends TaskProviderHelper implements TextToVideoTaskHelper {
 	constructor() {
 		super("novita", NOVITA_API_BASE_URL);
 	}
 
-	makeRoute(params: UrlParams): string {
-		return `/v3/hf/${params.model}`;
+	override makeRoute(params: UrlParams): string {
+		if (params.authMethod !== "provider-key") {
+			return `/v3/async/${params.model}?_subdomain=queue`;
+		}
+		return `/v3/async/${params.model}`;
 	}
 
-	preparePayload(params: BodyParams): Record<string, unknown> {
+	override preparePayload(params: BodyParams): Record<string, unknown> {
+		const { num_inference_steps, ...restParameters } = params.args.parameters as Record<string, unknown>;
 		return {
 			...omit(params.args, ["inputs", "parameters"]),
-			...(params.args.parameters as Record<string, unknown>),
+			...restParameters,
+			steps: num_inference_steps,
 			prompt: params.args.inputs,
 		};
 	}
```

Comment:
No need for casting if you use the proper type here
Also, `params.args.parameters` can be undefined

```suggestion
	override preparePayload(params: BodyParams<TextToVideoArgs>): Record<string, unknown> {
		const { num_inference_steps, ...restParameters } = params.args.parameters ?? {};
		return {
			...omit(params.args, ["inputs", "parameters"]),
			...restParameters,
			steps: num_inference_steps,
			prompt: params.args.inputs,
		};
	}
```

Reply:
Hi @SBrandeis, thanks for this suggestion. And I will encounter the following error if I don't cast the type explicitly:

```
Property 'num_inference_steps' does not exist on type '{}'.ts(2339)
```

Reply:
and I'm refactoring the code to handle cases where params.args.parameters is null or undefined by using an empty object {} as the default value. This ensures the destructuring operation won't fail even if the parameters property doesn't exist:

```typescript 
const { num_inference_steps, ...restParameters } = (params.args.parameters as Record<string, unknown>) ?? {};
```